code/datums/abilities/wraith.dm

```diff
@@ -1597,84 +1597,90 @@ ABSTRACT_TYPE(/datum/targetable/wraithAbility/curse)
 /datum/targetable/wraithAbility/create_summon_portal
 	name = "Summon void portal"
 	icon_state = "open_portal"
-	desc = "Summon a void portal from which otherworldly creatures pour out. You get increased point generation when near it."
-	pointCost = 150
+	desc = "Summon a void portal from which otherworldly creatures pour out. You get increased point generation when near it. Use ability again to upgrade it or destroy it."
+	pointCost = 100
 	targeted = 0
-	cooldown = 3 MINUTES
-	var/list/mob_types = list("Bears",
-	"Brullbars",
-	"Crunched",
-	"Ancient things",
-	"Ancient repairbots",
-	"Heavy gunner drones",
-	"Monstrosity crawlers",
-	"Bats",
-	"Shades",
-	"Lions",
-	"Skeletons",
-	"Random")
+	cooldown = 120 SECONDS
 
 	cast()
 		if (..())
-			return 1
+			return TRUE
 
 		var/turf/T = get_turf(holder.owner)
 		if (isturf(T) && istype(T,/turf/simulated/floor))
+			for (var/turf/simulated/wall/W in T)
+				boutput(holder.owner, "<span class='notice'>You cannot place your portal in a wall!</span>")
+				return TRUE
+			for (var/obj/window/W in T)
```

Comment:
Are you sure just windows? Don't we want to check for like all dense objects or something?

Reply:
I'll now check for objects with density = 1